File: process_sig.py
import os
import cv2
import numpy as np

photo_dir = r"c:\Users\Rocket\OneDrive\Desktop\portfolio\Photos"
out_dir = r"c:\Users\Rocket\OneDrive\Desktop\portfolio\fares-portfolio\public\images"

for file in os.listdir(photo_dir):
    if not file.endswith(".jpeg"): continue
    
    path = os.path.join(photo_dir, file)
    img = cv2.imread(path, cv2.IMREAD_GRAYSCALE)
    if img is None: continue
    
    # check if it's a signature (mostly white)
    mean_val = np.mean(img)
    if mean_val > 200: # mostly white
        print("Found signature:", file)
        
        # Convert to BGRA
        img_color = cv2.cvtColor(img, cv2.COLOR_GRAY2BGRA)
        
        # Make white transparent, and black -> white
        # We want the ink to be white, and background transparent
        # Threshold: anything below 150 is ink
        _, mask = cv2.threshold(img, 150, 255, cv2.THRESH_BINARY_INV)
        
        # Set all pixels to white
        img_color[:, :, 0] = 255
        img_color[:, :, 1] = 255
        img_color[:, :, 2] = 255
        # Set alpha channel to the mask (ink becomes opaque white, rest is transparent)
        img_color[:, :, 3] = mask
        
        out_path = os.path.join(out_dir, "signature.png")
        cv2.imwrite(out_path, img_color)
        print("Saved to", out_path)
        break

File: fares-portfolio/process_sig.js
const fs = require('fs');
const path = require('path');
const { Jimp } = require('jimp');

const photoDir = path.join(__dirname, '../Photos');
const outDir = path.join(__dirname, 'public/images');

async function processSignature() {
  const files = fs.readdirSync(photoDir).filter(f => f.endsWith('.jpeg') || f.endsWith('.jpg') || f.endsWith('.png'));
  
  for (const file of files) {
    const filePath = path.join(photoDir, file);
    try {
      const image = await Jimp.read(filePath);
      
      // Check if it's mostly white to identify the signature
      let whiteCount = 0;
      let totalCount = image.bitmap.width * image.bitmap.height;
      
      // Simple sampling
      for(let x=0; x<image.bitmap.width; x+=10) {
        for(let y=0; y<image.bitmap.height; y+=10) {
          const color = image.getPixelColor(x, y);
          const r = (color >> 24) & 255;
          const g = (color >> 16) & 255;
          const b = (color >> 8) & 255;
          if (r > 200 && g > 200 && b > 200) whiteCount++;
        }
      }
      
      if (whiteCount / (totalCount/100) > 0.6) {
        console.log("Found signature:", file);
        
        // Process
        image.scan(0, 0, image.bitmap.width, image.bitmap.height, function(x, y, idx) {
          const r = this.bitmap.data[idx + 0];
          const g = this.bitmap.data[idx + 1];
          const b = this.bitmap.data[idx + 2];
          
          const avg = (r + g + b) / 3;
          
          if (avg < 150) {
            // Ink -> Make it white
            this.bitmap.data[idx + 0] = 255;
            this.bitmap.data[idx + 1] = 255;
            this.bitmap.data[idx + 2] = 255;
            this.bitmap.data[idx + 3] = 255 - avg; // Anti-aliasing
          } else {
            // Background -> Make it transparent
            this.bitmap.data[idx + 0] = 255;
            this.bitmap.data[idx + 1] = 255;
            this.bitmap.data[idx + 2] = 255;
            this.bitmap.data[idx + 3] = 0;
          }
        });
        
        const outPath = path.join(outDir, 'signature.png');
        await image.write(outPath);
        console.log("Saved to", outPath);
        break;
      }
    } catch (e) {
      console.error(e);
    }
  }
}

processSignature();
